### PR TITLE
Refactor router navigation

### PR DESCRIPTION
**Pull request by btford**
This PR includes a couple commits:
- A quick rename
- A sizable refactor of the router's navigation control flow
- An improvement to how the route recognition prioritizes routes when matching

I'd normally split these into separate PRs, but ee4c1d0 depends heavily on d04b945.

### CODE REVIEW

File: modules/angular2/src/router/path_recognizer.js [hunk 1]
```diff
@@ -69,17 +69,16 @@ function parsePathString(route:string) {
...
       ListWrapper.push(results, new DynamicSegment(match[1]));
-      cost += 100;
+      specificity += (100 - i);
     } else if (isPresent(match = RegExpWrapper.firstMatch(wildcardMatcher, segment))) {
       ListWrapper.push(results, new StarSegment(match[1]));
-      cost += 10000;
     } else if (segment.length > 0) {
       ListWrapper.push(results, new StaticSegment(segment));
-      cost += 1;
+      specificity += 100 * (100 - i);
```
Comment (rkirov) [inline modules/angular2/src/router/path_recognizer.js]: I see what you are doing with specificity here, but I think it is non-trivial enough to deserve a comment.

File: modules/angular2/src/router/pipeline.js [hunk 2]
```diff
@@ -10,15 +10,6 @@ export class Pipeline {
   steps:List;
   constructor() {
     this.steps = [
```
Comment (rkirov) [inline modules/angular2/src/router/pipeline.js]: Is a pipeline of one-step even needed?

File: modules/angular2/src/router/router.js [hunk 3]
```diff
@@ -94,9 +97,8 @@ export class Router {
   config(config:any) {
     if (config instanceof List) {
       config.forEach((configObject) => {
-        // TODO: this is a hack
```
Comment (rkirov) [inline modules/angular2/src/router/router.js]: no longer a hack, huh :)

File: modules/angular2/src/router/router.js [hunk 4]
```diff
@@ -208,17 +232,26 @@ export class Router {
...
+    super(registry, pipeline, null, hostComponent);
+    this._location = location;
     this._location.subscribe((change) => this.navigate(change['url']));
     this._registry.configFromComponent(hostComponent);
     this.navigate(location.path());
   }
+
+  commit(instruction):Promise {
+    return super.commit(instruction).then((_) => {
+      this._location.go(instruction.accumulatedUrl);
```
Comment (rkirov) [inline modules/angular2/src/router/router.js]: why is the root router navigating, isn't that what `navigate` is for?

File: modules/angular2/src/router/router.js (hunk 3 above)
Comment (btford) [inline modules/angular2/src/router/router.js]: It graduated to "legit code" a few revisions ago – the comment is already out of date.

File: modules/angular2/src/router/pipeline.js (hunk 2 above)
Comment (btford) [inline modules/angular2/src/router/pipeline.js]: If you look closely at the other existing steps, all of them are basically stubbed out as `PromiseWrapper.resolve(true)`. I'll reintroduce those steps in a follow-up PR.

File: modules/angular2/src/router/router.js (hunk 4 above)
Comment (btford) [inline modules/angular2/src/router/router.js]: The root router should be the one to update the URL.

**Comment by btford** [comment]
Landed as 6b02cb9b446f0f4b70061affb09c6c65434a040c, c29ab86d857556f5549e46780694963574f8a322, 5db89071d408c2e5b2e26b3d7b00e02e164c197f, and 83b97c485ba3ed80d9155150e2300d0dad192187.

**Pull request closed without merging**